mtags/src/main/scala-3/scala/meta/internal/pc/printer/DotcPrinter.scala

```diff
@@ -67,6 +70,24 @@ object DotcPrinter:
     def fullName(sym: Symbol): String =
       fullNameString(sym)
 
+    override protected def toTextRefinement(rt: RefinedType): Closed =
+      rt.refinedInfo match
+        case TypeBounds(lo, hi) if lo == hi =>
+          val tsym = rt.parent.member(rt.refinedName).symbol
+          if tsym.exists then
+            ("type " ~ simpleNameString(tsym) ~ " = " ~ toText(hi)).close
+          else super.toTextRefinement(rt)
+        case tp: TypeRef =>
```

Comment:
In the last iteration I also added support for including def / val/ var, which is needed with Selectable